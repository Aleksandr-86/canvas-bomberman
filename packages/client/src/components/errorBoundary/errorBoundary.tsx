import { Component, ReactNode } from 'react'

import { ErrorStub } from '../errorStub/errorStub'

interface Props {
  children?: ReactNode
}

interface State {
  hasError: boolean
}

export class ErrorBoundary extends Component<Props, State> {
  public state: State = {
    hasError: false,
  }

  public static getDerivedStateFromError(): State {
    return { hasError: true }
  }

  public render() {
    if (this.state.hasError) {
      return <ErrorStub />
    }

    return this.props.children
  }
}
