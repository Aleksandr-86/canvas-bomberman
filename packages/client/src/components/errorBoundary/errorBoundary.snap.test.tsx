import * as renderer from 'react-test-renderer'
import { ErrorBoundary } from './errorBoundary'

const mockChildrenElement = <div>mockChildrenElement</div>

describe(`Snapshot test component ErrorBoundary`, () => {
  test(`Should correct render (with props)`, () => {
    const tree = renderer
      .create(<ErrorBoundary>{mockChildrenElement}</ErrorBoundary>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
