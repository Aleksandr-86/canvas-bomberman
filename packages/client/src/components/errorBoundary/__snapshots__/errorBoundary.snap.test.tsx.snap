// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test component ErrorBoundary Should correct render (with props) 1`] = `
<div>
  mockChildrenElement
</div>
`;
